awx/main/models/events.py

```diff
@@ -575,12 +575,26 @@ def _update_host_summary_from_stats(self, hostnames):
 
             Host.objects.bulk_update(list(updated_hosts), ['last_job_id', 'last_job_host_summary_id'], batch_size=100)
 
-            # bulk-create
-            current_time = now()
-            HostMetric.objects.bulk_create(
-                [HostMetric(hostname=hostname, last_automation=current_time) for hostname in updated_hosts_list], ignore_conflicts=True, batch_size=100
+            # Create/update Host Metrics
+            self._update_host_metrics(updated_hosts_list)
+
+    @staticmethod
+    def _update_host_metrics(updated_hosts_list):
+        from awx.main.models import HostMetric  # circular import
+
+        # bulk-create
+        current_time = now()
+        HostMetric.objects.bulk_create(
+            [HostMetric(hostname=hostname, last_automation=current_time) for hostname in updated_hosts_list], ignore_conflicts=True, batch_size=100
+        )
+        # bulk-update
+        batch_start, batch_size = 0, 1000
+        while batch_start <= len(updated_hosts_list):
```

Comment:
It looks like with `<=` you could end up with an extra process. For example:
```
>>> for item in range(0,30):
...     array.append( item )
>>> batch_start, batch_size = 0, 10
>>> while batch_start <= len(array):
...     batched_host_list = array[batch_start : (batch_start + batch_size)]
...     print(batched_host_list)
...     batch_start += batch_size
... 
[0, 1, 2, 3, 4, 5, 6, 7, 8, 9]
[10, 11, 12, 13, 14, 15, 16, 17, 18, 19]
[20, 21, 22, 23, 24, 25, 26, 27, 28, 29]
[]
```

I don't think it would hurt anything to have that extra empty list at the end from a quick look.